driver/src/test/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphIndexTest.java

```diff
@@ -106,26 +113,67 @@ public void vertexUniqueIndexLookupWithValue() {
     }
 
     @Test
+    public void vertexUniqueIndexLookupWithMultipleLabels() {
+        final String label1 = "label1";
+        final String label2 = "label2";
+        final String label3 = "label3";
+
+        final String value1 = "value1";
+
+        OrientGraph graph = newGraph();
+        createVertexIndexLabel(graph, label1);
+        createVertexIndexLabel(graph, label2);
+        createVertexIndexLabel(graph, label3);
+
+        //Check that property (key) is indexed on multiple labels
+        Assert.assertEquals(graph.getIndexedKeys(Vertex.class, label1), new HashSet<>(Collections.singletonList(key)));
+        Assert.assertEquals(graph.getIndexedKeys(Vertex.class, label2), new HashSet<>(Collections.singletonList(key)));
+        Assert.assertEquals(graph.getIndexedKeys(Vertex.class, label3), new HashSet<>(Collections.singletonList(key)));
+
+        Vertex v1 = graph.addVertex(label, label1, key, value1);
+        Vertex v2 = graph.addVertex(label, label2, key, value1);
+        Vertex v3 = graph.addVertex(label, label3, key, value1);
+
+        GraphTraversal<Vertex, Vertex> traversal = graph.traversal().V().hasLabel(label1, label2, label3).has(key, value1);
+
+        Set<OrientIndexQuery> indicies = findUsedIndex(traversal);
+        Assert.assertEquals(3, indicies.size());
+
+        assertTrue(valueFound(indicies, v1, value1));
```

Comment:
we would we get better test assertion errors if you use `assertEquals`.. you could e.g. rename `valueFound` to `assertValueFound` and use `assertEquals` in there. 

just a thought..